### PR TITLE
perf(export): only call FindEnvs once

### PR DESCRIPTION
**Pull request by Duologic**
On the internal Grafana Labs jsonnet code base, this can improve the performance of the export significantly. Tests on my local
machine went from 2m08s to 1m32s.

### CODE REVIEW

File: pkg/tanka/parallel.go [hunk 1]
```diff
@@ -92,6 +80,7 @@ type parallelOut struct {
 
 func parallelWorker(jobsCh <-chan parallelJob, outCh chan parallelOut) {
 	for job := range jobsCh {
+		log.Printf("Loading %s from %s", job.opts.Name, job.path)
```
Comment (sh0rez) [inline pkg/tanka/parallel.go:83]: Supposed to be here? Sounds quite verbose imo

File: pkg/tanka/export.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ type ExportEnvOpts struct {
 	Parallelism int
 }
 
-func ExportEnvironments(paths []string, to string, opts *ExportEnvOpts) error {
+func ExportEnvironments(paths map[string]string, to string, opts *ExportEnvOpts) error {
```
Comment (sh0rez) [inline pkg/tanka/export.go:46]: It is quite unobvious what this `map[string]string` should hold. From looking at the code, it appears to be a name -> entrypoint mapping.

Wdyt about passing a `[]Environment` instead, where the elements are expected to be sparse environments (missing `data` key)?

**Review by sh0rez — CHANGES_REQUESTED** [review]
Nice work!

File: pkg/tanka/parallel.go (hunk 1 above)
Comment (Duologic) [reply pkg/tanka/parallel.go:83]: I left it intentionally. It gives a bit of feedback to the user as opposed to no feedback right now. I'm happy to remove or accept other suggestions to give a sense of progress.

Comment (sh0rez) [reply pkg/tanka/parallel.go:91]: Fair enough. I'm not totally familiar, does this codepath affect any other actions than export?

**Review by sh0rez — APPROVED** [review]

File: pkg/tanka/parallel.go (hunk 1 above)
Comment (Duologic) [reply pkg/tanka/parallel.go:91]: Not that I know off.

**Pull request merged**